planner/core/encode.go

```diff
@@ -64,5 +71,95 @@ func (pn *planEncoder) encodePlan(p PhysicalPlan, isRoot bool, depth int) {
 	case *PhysicalIndexLookUpReader:
 		pn.encodePlan(copPlan.indexPlan, false, depth)
 		pn.encodePlan(copPlan.tablePlan, false, depth)
+	case *PhysicalIndexMergeReader:
+		for i := 0; i < len(copPlan.partialPlans); i++ {
+			pn.encodePlan(copPlan.partialPlans[i], false, depth)
+		}
+		if copPlan.tablePlan != nil {
+			pn.encodePlan(copPlan.tablePlan, false, depth)
+		}
+	}
+}
+
+var digesterPool = sync.Pool{
+	New: func() interface{} {
+		return &planDigester{
+			hasher: sha256.New(),
+		}
+	},
+}
+
+type planDigester struct {
+	buf          bytes.Buffer
+	encodedPlans map[int]bool
+	hasher       hash.Hash
+}
+
+// NormalizePlan is used to normalize the plan and generated plan digest.
+func NormalizePlan(p Plan) (normalized, digest string) {
+	d := digesterPool.Get().(*planDigester)
+	defer digesterPool.Put(d)
+	selectPlan := getSelectPlan(p)
+	if selectPlan == nil {
+		return "", ""
```

Comment:
Does the tests cover this branch?

Reply:
Ye, the insert test will cover this. https://github.com/pingcap/tidb/pull/13124/files#diff-825bf3080efc8eba2fa560056b3b8997R135